sympy/core/basic.py

```diff
@@ -929,23 +929,10 @@ def subs(self, *args, **kwargs):
 
         if unordered:
             sequence = dict(sequence)
-            if not all(k.is_Atom for k in sequence):
-                d = {}
-                for o, n in sequence.items():
-                    try:
-                        ops = o.count_ops(), len(o.args)
-                    except TypeError:
-                        ops = (0, 0)
-                    d.setdefault(ops, []).append((o, n))
-                newseq = []
-                for k in sorted(d.keys(), reverse=True):
-                    newseq.extend(
-                        sorted([v[0] for v in d[k]], key=default_sort_key))
-                sequence = [(k, sequence[k]) for k in newseq]
-                del newseq, d
-            else:
-                sequence = sorted([(k, v) for (k, v) in sequence.items()],
-                                  key=default_sort_key)
+            atoms, nonatoms = sift(list(sequence),
+                lambda x: x.is_Atom, binary=True)
+            sequence = [(k, sequence[k]) for k in
+                list(reversed(list(ordered(nonatoms)))) + list(ordered(atoms))]
```

Comment:
Rather than `list(reversed(list(...)))` you can do `list(...)[::-1]` which might be more efficient

Reply:
Sorry I missed that, @oscarbenjamin .
```python
>>> timeit('list(reversed([1,2,3,4,5,6,7,8,9,0]))')
2.58999991417
>>> timeit('[1,2,3,4,5,6,7,8,9,0][::-1]')
0.996271133423
```